### PR TITLE
Initial text for SLSA chapter

### PR DESCRIPTION
**Pull request by alextserepov**
- Levels of assurance

Signed-off-by: Aleksandr Tserepov-Savolainen <alextserepov@gmail.com>

### CODE REVIEW

File: src/scs/slsa-framework.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# SLSA Framework
+
+Supply chain Levels for Software Artifacts (SLSA) is a security framework for tampering prevention, integrity improvement and securing packages and infrastructure of a project.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:3]: ... tampering prevention, integrity improvement, and securing packages and infrastructure of a project.

Should we add this link https://slsa.dev/ somewhere?

File: src/scs/slsa-framework.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Levels of assurance
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:5]: Levels of Assurance

File: src/scs/slsa-framework.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+One of the requirements for the solution is to reach SLSA Level 4 and even go beyond that. This requires a lot of process changes as well as technical work. Current status for Thincompute project is visualized at the separate wiki page: Supply Chain Security
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:7]: Currently, the **Ghaf** project is visualized on...
@brianmcgillion am I right that here we do not want to use Thincompute as a project? I replaced it with Ghaf.

Comment (jenninikko) [inline src/scs/slsa-framework.md:7]: ... the separate wiki page: Supply Chain Security 

I think we should not use internal links here. To what do you want to refer here?

File: src/scs/slsa-framework.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# SLSA Framework
+
+Supply chain Levels for Software Artifacts (SLSA) is a security framework for tampering prevention, integrity improvement and securing packages and infrastructure of a project.
+
+## Levels of assurance
+
+One of the requirements for the solution is to reach SLSA Level 4 and even go beyond that. This requires a lot of process changes as well as technical work. Current status for Thincompute project is visualized at the separate wiki page: Supply Chain Security 
+
+The SLSA model consists of 4 levels, offering incremental level of anti-tampering protection. Levels 1-3 represent milestones with certain integrity guarantees, whereas level 4 is an ideal end state.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:9]: ... offering **an** incremental level of anti-tampering protection.

Comment (jenninikko) [inline src/scs/slsa-framework.md:9]: Oh, and if we have this description here, can I remove this one scs/slsa-build-system.md?

File: src/scs/slsa-framework.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+**Level 0** means no SLSA compliance and no guarantees are given.
+
+**Level 1** demands fully scripted/automated build process and provenance generation. It offers basic confidence of the origin of the software but doesn’t provide any tamper resistance.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:13]: ...demands **a** fully scripted/automated...

... confidence **in** the origin...

File: src/scs/slsa-framework.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+**Level 2** Introduces authenticated provenance generated by a hosted build at the same time demanding version control system usage. Provenance prevents tampering and offers trust in build service.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:15]: ...in **the** build service.

File: src/scs/slsa-framework.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+**Level 3** offers auditability of the source and and the integrity of the provenance. Much stronger tampering protection is provided by preventing specific classes of threats such as cross-build contamination.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:17]: ... of the source **and** (one extra and) the integrity of the provenance.

File: src/scs/slsa-framework.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+**Level 4** requires two-peer source code review and a hermetic, reproducible build process. Overall Level 4 offers high degree of confidence in anti-tampering protection of the software.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:19]: ...offers **a** high degree of confidence in **the** anti-tampering protection...

File: src/scs/slsa-framework.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+Important note is that SLSA level is not transitive, thus level of the artifact is not dependent on the level of dependencies, which are expected to have their own SLSA levels. This makes it possible to build Level 4 artifact from Level 0 dependencies.
```
Comment (jenninikko) [inline src/scs/slsa-framework.md:21]: Should we remove "Important note is that" and start with: SLSA level is not transitive...


...to build Level 4 artifact**s** from Level 0 dependencies. 
Or if you want to use a singular here, then: ...build **a** Level 4 artifact...

**Review by jenninikko — CHANGES_REQUESTED** [review]
@alextserepov my comments are below <3

File: src/scs/slsa-framework.md (hunk 1 above)
Comment (alextserepov) [reply src/scs/slsa-framework.md:3]: Yes, the link should (must) be added, but where to?

File: src/scs/slsa-framework.md (hunk 4 above)
Comment (alextserepov) [reply src/scs/slsa-framework.md:9]: good question. Let's think about it a bit.

File: src/scs/slsa-framework.md (hunk 3 above)
Comment (brianmcgillion) [reply src/scs/slsa-framework.md]: correct

File: src/scs/slsa-framework.md (hunk 1 above)
Comment (jenninikko) [reply src/scs/slsa-framework.md]: Ok, let's think about it when others sections will be ready.

File: src/scs/slsa-framework.md (hunk 4 above)
Comment (jenninikko) [reply src/scs/slsa-framework.md]: I well remove it after the merge.

**Pull request merged**